src/consumer/runner.js

```diff
@@ -344,27 +340,19 @@ module.exports = class Runner {
 
               await onBatch(batch)
               await this.consumerGroup.heartbeat({ interval: this.heartbeatInterval })
-            } catch (e) {
-              unlockWithError(e)
-            } finally {
-              numberOfExecutions++
-              if (numberOfExecutions === expectedNumberOfExecutions) {
-                unlock()
-              }
-            }
-          }).catch(unlockWithError)
-        )
+            })
+          )
+        }
       })
     }
 
+    const enqueuedTasks = await Promise.all(allResults.map(fn => fn()))
+
     if (!this.running) {
       return
     }
 
-    await Promise.all(enqueuedTasks.map(fn => fn()))
-    if (expectedNumberOfExecutions > 0) {
-      await lock
-    }
+    await Promise.all([].concat(...enqueuedTasks.map(fn => fn())))
```

Comment:
I am using `[].concat(...enqueuedTasks)` to flatten `enqueuedTasks` as enqueuedTasks looks like [[batch, batch, batch], [batch, batch]] above one will flatten it and we will await for every single batch to complete of cause taking in account `concurrently` logic